### ISSUE TITLE
[Feature request] Visual Studio IDE's ToolBox alternative for VSCode

### ISSUE THREAD

**Issue by BachiMjavanadze** [carlocardella/vscode-TextToolbox#35]
`Visual Studio IDE` has a nice feature: ToolBox, witch gives you ability to save and use pieces of boilerplate code:

![tImg](https://user-images.githubusercontent.com/38082501/213792825-2ff83c48-3655-4e6d-94d4-31bcf1573709.gif)

Can't find a similar extension for `VSCode`. There many extensions with the similar name but they serve different job and are a bit complicated for me.

Can you add similar feature to your extention please?

P.S. I' not asking about snippets, this is different.

**Comment by BachiMjavanadze** [comment]
Almost find it but it does not have foldering option and is no longer updated:
[https://marketplace.visualstudio.com/items?itemName=markvincze.code-fragments](url)

**Comment by carlocardella** [comment]
I have not used Visual Studio in a while but this looks interesting.
Can you give me a better idea of what you are looking for? For example, is a "generic" list of _code blocks_ (for lack of a better term) or would you want them separated by language?

I was actually thinking about a similar functionality, I am still trying to decide if it would fit in this extension or should be a separate one.

Off the top of my head I could think of a few points:
1. Create custom "toolboxes" (e.g. per language or other custom groups)
2. Sort, grouping etc.
3. Possibly, sync across devices
4. What else?

**Comment by BachiMjavanadze** [comment]
I think it woud be better if we have both ability: separate snippets by language and have general ones. Main part is: ability to comain snippets in diifrent folders even in single language. i.e. for C# I will have diffrent groups for "console", "asp.net", "MAUI".
 
 `sync across devices` must have!

**Comment by carlocardella** [comment]
> `sync across devices` must have!

Yeah, in my list I said "possibly" because it is simple to sync settings and custom values, but in this case the amount of data to sync is potentially high and there is no way, that I know of, to sync files (I can see those code blocks to be saved on a file for the extension, but those files are not shared with the regular "Settings sync" feature).
I'll need to investigate if there are size limits; I know of other extension authors that have asked for this feature in the VSCode repo but it is still not possible. 

Anyway, looks like I have something to think about 😄

**Comment by BachiMjavanadze** [comment]
I think adding folding option per language will be tricky. maybe the easiest solution is to add subfolders and user will define folder for languages. i.e: `.Net` will be parent folder and subfolders will be: `console` and `WebApi`

**Comment by carlocardella** [comment]
@BachiMjavanadze I've been thinking about this, can you please clarify why you do not want to use snippets? 

This is how I see it so far:
1. Snippets are easily synchronized across machines with the built-in settings sync
2. Snippets do not have a GUI at the moment, I am thinking that this could be a sort of "snippet manager extension" (that was the new extension I had been thinking about anyway)
3. Custom snippets are a bit cumbersome to create because of the json escaping, an extension could help (just select the text you want to transform into a snippet and the extension automnatically converts and saves it in the proper snippet file)
4. The grouping could be easily achieved with the TreeView control: groups (or "folders") can be logical containers used to group snippets from across the underlying files
5. The drawback would be that these snippets would likely how up as regular snippets (unless it is possible to disable specific snippet categories, I need to look into it)

Thoughts?

**Comment by BachiMjavanadze** [comment]
`why you do not want to use snippets?`
Who sex that I do not want to use snippets?! Actualy I use snippets very often but the toolbox is diffrent; It's for boilerplate code. I can put many lines of boilerplate code. `Visual Studio IDE` has both ones: snippets and toolbox and this is very useful.

You may mask underhood the tooltip like a regular snippet and they will be easily synchronized

**Comment by carlocardella** [comment]
I asked about snippets because of this:

> P.S. I' not asking about snippets, this is different.

I wanted to understand if you do not want snippets at all and if so, what is the drawback you see.

This clarifies, thanks
